src/Compression/CompressionCodecDeflate.cpp

```diff
@@ -0,0 +1,357 @@
+#include <cstdio>
+#include <thread>
+#include <Compression/CompressionCodecDeflate.h>
+#include <Compression/CompressionFactory.h>
+#include <Compression/CompressionInfo.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Poco/Logger.h>
+#include <Common/logger_useful.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int CANNOT_COMPRESS;
+    extern const int CANNOT_DECOMPRESS;
+}
+//DeflateJobHWPool
+qpl_job * DeflateJobHWPool::jobPool[jobPoolSize];
+std::atomic_bool DeflateJobHWPool::jobLock[jobPoolSize];
+
+DeflateJobHWPool & DeflateJobHWPool::instance()
+{
+    static DeflateJobHWPool ret;
+    return ret;
+}
+
+DeflateJobHWPool::DeflateJobHWPool()
+{
+    log = &Poco::Logger::get("DeflateJobHWPool");
+    if (initJobPool() < 0)
+    {
+        jobPoolEnabled = false;
+        LOG_WARNING(log, "DeflateJobHWPool is not ready. Please check if IAA hardware support.Auto switch to deflate software codec here");
```

Comment:
"Initialization of hardware-assisted DEFLATE codec failed, falling back to software DEFLATE codec. Please check if Intel In-Memory Analytics Accelerator (IAA) is properly set up."

Reply:
Done